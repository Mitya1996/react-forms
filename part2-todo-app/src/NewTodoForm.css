.Form {
  margin-left: 20px;
}
