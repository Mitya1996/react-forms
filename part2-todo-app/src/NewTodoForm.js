import { useForm } from "react-hook-form";
import { v4 as uuid } from "uuid";
import { yupResolver } from "@hookform/resolvers/yup";
import * as yup from "yup";
import { Button, TextField } from "@material-ui/core";
import "./NewTodoForm.css";

const schema = yup.object().shape({
  task: yup.string().max(50).required(),
});

function NewTodoForm({ createTodo }) {
  const {
    register,
    handleSubmit,
    formState: { errors },
  } = useForm({
    resolver: yupResolver(schema),
  });
  const onSubmit = (data) => createTodo({ ...data, id: uuid() });

  return (
    <form className="Form" onSubmit={handleSubmit(onSubmit)}>
      <TextField
        {...register("task")}
        helperText={errors.task?.message}
        error={errors.task}
        variant="outlined"
        label="Task"
      />

      <div style={{ margin: "10px 0 0 20px" }}>
        <Button type="submit" variant="contained">
          Add Todo
        </Button>
      </div>
    </form>
  );
}

export default NewTodoForm;
