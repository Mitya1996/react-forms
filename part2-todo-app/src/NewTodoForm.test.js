import NewTodoForm from "./NewTodoForm";
import { render, fireEvent } from "@testing-library/react";

//smoke test
it("renders without crashing", function () {
  render(<NewTodoForm />);
});

// snapshot test
it("matches snapshot", function () {
  const { asFragment } = render(<NewTodoForm />);
  expect(asFragment()).toMatchSnapshot();
});
