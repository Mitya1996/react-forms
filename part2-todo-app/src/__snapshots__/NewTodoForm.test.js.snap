// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <form
    class="Form"
  >
    <div
      class="MuiFormControl-root MuiTextField-root"
    >
      <label
        class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
        data-shrink="false"
      >
        Task
      </label>
      <div
        class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
      >
        <input
          aria-invalid="false"
          class="MuiInputBase-input MuiOutlinedInput-input"
          name="task"
          type="text"
          value=""
        />
        <fieldset
          aria-hidden="true"
          class="PrivateNotchedOutline-root-5 MuiOutlinedInput-notchedOutline"
        >
          <legend
            class="PrivateNotchedOutline-legendLabelled-7"
          >
            <span>
              Task
            </span>
          </legend>
        </fieldset>
      </div>
    </div>
    <div
      style="margin: 10px 0px 0px 20px;"
    >
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-contained"
        tabindex="0"
        type="submit"
      >
        <span
          class="MuiButton-label"
        >
          Add Todo
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </div>
  </form>
</DocumentFragment>
`;
